api/rosetta/errors.go

```diff
@@ -33,94 +32,106 @@ type Error struct {
 	Details     map[string]interface{} `json:"details,omitempty"`
 }
 
-func Internal(err error) Error {
-	return Error{
-		ErrorDefinition: configuration.ErrorInternal,
-		Description:     err.Error(),
-		Details:         nil,
+type OptionFunc func(*Error)
```

Comment:
I changed it to `Detail`, `OptionFunc` was weird for me too, I intended to change it as the draft PR went closer to a final PR.

Do you think that `Detail` name is clear though? Especially with how I did it, where it takes an `*Error` argument, it's perhaps not clear by name what it does..

I was thinking of `Detailer` or `Decorator`, or perhaps having it use `map[string]interface{}` as an argument instead of the `*Error`. Not sure which is nicest

Reply:
Yeah maybe Detailer is better indeed 👍 (_Decorator_ reminds me of Java and makes my skin crawl though hehe)

Reply:
`Detailer` is weird, `Detail` works.